Src/HALAL/Services/Communication/I2C/I2C.cpp

```diff
@@ -0,0 +1,248 @@
+/*
+ * SPI.hpp
+ *
+ *  Created on: 5 nov. 2022
+ *      Author: Pablo
+ */
+#include "Communication/I2C/I2C.hpp"
+
+#ifdef HAL_I2C_MODULE_ENABLED
+
+unordered_map<uint8_t, I2C::Instance* > I2C::active_i2c;
+
+uint16_t I2C::id_counter = 0;
+
+optional<uint8_t> I2C::inscribe(I2C::Peripheral& i2c){
+	if ( !I2C::available_i2cs.contains(i2c)){
+
+		return nullopt; //TODO: Error handler or throw the exception
+	}
+
+	I2C::Instance* i2c_instance = I2C::available_i2cs[i2c];
+
+    Pin::inscribe(i2c_instance->SCL, ALTERNATIVE);
+    Pin::inscribe(i2c_instance->SDA, ALTERNATIVE);
+
+    DMA::inscribe_stream(i2c_instance->RX_DMA);
+    DMA::inscribe_stream(i2c_instance->TX_DMA);
+
+    uint8_t id = I2C::id_counter++;
+
+    I2C::active_i2c[id] = i2c_instance;
+
+    return id;
+}
+
+
+void I2C::start(){
+	for(auto iter: I2C::active_i2c){
+		I2C::init(iter.second);
+	}
+}
+
+
+bool I2C::transmit_next_packet(uint8_t id, I2CPacket& packet){
+    if (!I2C::active_i2c.contains(id)){
+    	ErrorHandler("I2C No encontrado al enviar paquete \n\r");
+        return false; //TODO: Error handler
+    }
+
+    I2C::Instance* i2c = I2C::active_i2c[id];
+
+
+    if(i2c->hi2c->State == HAL_I2C_STATE_BUSY_TX){
+    	ErrorHandler("I2C Transmit buffer busy!\n\r");
+    	return false;
+    }
+
+    if (HAL_I2C_Master_Transmit_DMA(i2c->hi2c, packet.get_id(), packet.get_data(), packet.get_size()) != HAL_OK){
+    	ErrorHandler("Error during I2C transmission\n\r");
+        return false; //TODO: Warning, Error during transmision
+    }
+
+    printf("Packet send successfully \n\r");
```

Comment:
```suggestion
    printf("Packet send successfully \n\r");
```
Los prints son solo para debug, no se ponen en la libreria

Reply:
tengo que acordarme de borrarlo cuando termine de debuggear, me dejo esta abierta